sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceCleaner.java

```diff
@@ -1,44 +1,25 @@
 package com.mesosphere.sdk.offer;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-
-import org.apache.mesos.Protos;
 import org.apache.mesos.Protos.Offer;
 import org.apache.mesos.Protos.Resource;
-import com.mesosphere.sdk.state.StateStore;
-import com.mesosphere.sdk.state.StateStoreException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
 /**
  * The Resource Cleaner provides recommended operations for cleaning up
  * unexpected Reserved resources and persistent volumes.
  */
-public class ResourceCleaner {
+public abstract class ResourceCleaner {
```

Comment:
Given how little code is left here, it feels like ResourceCleaner could just be an interface to be implemented (independently) by DefaultResourceCleaner / UninstallResourceCleaner. The remaining piece in `evaluate()`, below, could be a separate static function to be called by DefaultResourceCleaner / UninstallResourceCleaner. I'm leaning towards this structure as it'll make the two a little more isolated and reduce the chance of unintentional changes from one side affecting the other, as ideally they don't really have much to do with each other.

TLDR:
- ResourceCleaner could be an interface
- DefaultResourceCleaner and UninstallResourceCleaner could implement that interface (but be independent otherwise)
- The common `evaluate()` code currently in ResourceCleaner could be moved into a static method to be invoked by DefaultResourceCleaner/UninstallResourceCleaner.